.github/workflows/checks.yml

```diff
@@ -24,25 +24,101 @@ jobs:
     - name: Install Nix
       uses: DeterminateSystems/nix-installer-action@main
 
-    - name: Run MCR tests in nix environment
-      run: nix develop --command bash  -c "just mcr-contract-tests"
-
-    - name: Run MCR client tests in nix environment
-      run: CELESTIA_LOG_LEVEL=FATAL MOVEMENT_PREBUILT=false nix develop --command bash  -c "just mcr test.local -t=false"
-
     - name: Run Cargo Check in nix environment
       run: nix develop --command bash  -c "cargo check"  
```

Comment:
`cargo check` builds in incremental mode by default.

This means all crates, including depenencies, will need to be built again by the test jobs.

It may be more efficient to do `cargo build --release` if that's what the test jobs are building with.
The full devil is in the details: some cargo flag thrown in by the test setup scripts may mean all crates will need to be rebuilt for the tests anyway. But disabling incremental builds is a worthwhile step to save some build time.